pkg/lockfile/parse-conan-lock.go

```diff
@@ -0,0 +1,177 @@
+package lockfile
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type ConanReference struct {
+	Name            string
+	Version         string
+	Username        string
+	Channel         string
+	RecipeRevision  string
+	PackageID       string
+	PackageRevision string
+	TimeStamp       string
+}
+
+type ConanGraphNode struct {
+	Pref      string `json:"pref"`
+	Ref       string `json:"ref"`
+	Options   string `json:"options"`
+	PackageId string `json:"package_id"`
+	Prev      string `json:"prev"`
+	Path      string `json:"path"`
+	Context   string `json:"context"`
+}
+
+type ConanGraphLock struct {
+	Nodes map[string]ConanGraphNode `json:"nodes"`
+}
+
+type ConanLockFile struct {
+	Version string `json:"version"`
+	// conan v0.4- lockfiles use "graph_lock", "profile_host" and "profile_build"
+	GraphLock    ConanGraphLock `json:"graph_lock,omitempty"`
+	ProfileHost  string         `json:"profile_host,omitempty"`
+	ProfileBuild string         `json:"profile_build,omitempty"`
+	// conan v0.5+ lockfiles use "requires", "build_requires" and "python_requires"
+	Requires       []string `json:"requires,omitempty"`
+	BuildRequires  []string `json:"build_requires,omitempty"`
+	PythonRequires []string `json:"python_requires,omitempty"`
+}
+
+const ConanEcosystem Ecosystem = "conan.io"
```

Comment:
Just flagging this isn't a valid ecosystem currently per the [OSV spec](https://ossf.github.io/osv-schema/#affectedpackage-field) - I'll leave it to @another-rex & @oliverchang to decide what to do about this

Reply:
Indeed! 

We should at the very least add a TODO here to say that this is tentative and subject to change depending on the OSV schema. 

This is definitely a useful addition to have for the OSV-Scanner, but the database itself also currently does not have conan.io advisories. Is there also an existing Conan vuln DB that we could have publishing OSV entries?

Reply:
> Is there also an existing Conan vuln DB that we could have publishing OSV entries?

Conan is the C++ package manager 😉 We have ConanCenter which offers a ton of C and C++ packages. https://cve.mitre.org/index.html or https://nvd.nist.gov/ are the types of places I'd expect find know vulnerabilities.

https://conan.io/center/openssl has numerous ones https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword=openssl.
[openssl/3.0.5](https://conan.io/center/openssl?version=3.0.5) has [CVE-2022-3602](https://nvd.nist.gov/vuln/detail/CVE-2022-3602) for example.

I saw in the readme [these are listed as alias](https://github.com/google/osv-scanner#output-format) could they not be the IDs as well? Some like the Openssl Alpine that I see on https://osv.dev/list?ecosystem=Alpine&q=openssl was my initial reaction.

Reply:
thanks, I've opened a PR to ovs-schema: https://github.com/ossf/osv-schema/pull/101
for the database, sorry, but I have no idea. maybe @prince-chrismc or @jcar87 know and can comment. perhaps, X-Ray might have some sort of database, but I am not familiar with it.